typing/ctype.mli

```diff
@@ -398,14 +406,33 @@ val get_current_level: unit -> int
 val wrap_trace_gadt_instances: Env.t -> ('a -> 'b) -> 'a -> 'b
 val reset_reified_var_counter: unit -> unit
 
-val immediacy : Env.t -> type_expr -> Type_immediacy.t
-
 val maybe_pointer_type : Env.t -> type_expr -> bool
        (* True if type is possibly pointer, false if definitely not a pointer *)
+val is_void_type : Env.t -> type_expr -> bool
 
 (* Stubs *)
 val package_subtype :
     (Env.t -> Path.t -> Longident.t list -> type_expr list ->
       Path.t -> Longident.t list -> type_expr list -> bool) ref
 
 val mcomp : Env.t -> type_expr -> type_expr -> unit
+
+val get_unboxed_type_representation : Env.t -> type_expr -> type_expr
```

Comment:
We may need to update our terminology. This function might look through `[@@unboxed]` things, or it might somehow get the layout of something. The fact that we use "unboxed" to refer to two notionally separate concepts is going to cause trouble, forever.

Inlined types?
Raw types?
Unpackaged types?

Maybe I'm overreacting, but I foresee many, many conversations that involve "Which 'unboxed' do you mean there?".

Reply:
I'd be interested to know how much `[@@unboxed]` is used.  If it's quite uncommon, maybe we don't need to panic.

(Note that this code isn't actually added by my PR, it's just showing up in the diff because is moved in upstream PR #10041, which has been pulled in here).